File: files/2026-02-01/CSEC 01022026.py
import pandas as pd

path = "2026-02-01\\Challenge 101.xlsx"
input1 = pd.read_excel(path, usecols= "B", nrows = 3, skiprows = 2)
input2 = pd.read_excel(path, usecols= "D:E", nrows = 2, skiprows = 1)
test = pd.read_excel(path, usecols= "G", nrows = 7, skiprows = 1)

result = pd.DataFrame({
    'items': [item for i, item in enumerate(input1['Items']) 
              for _ in range(input2.loc[input2['Item:'] == i+1, 'Repeats:'].values[0] 
                           if (i+1) in input2['Item:'].values else 1)]
})

print(result['items'].equals(test['Solution']))
# True

File: files/2026-02-01/CSEC 01022026.R
library(tidyverse)
library(readxl)

path <- "2026-02-01/Challenge 101.xlsx"
input1 <- read_excel(path, range = "B3:B6")
input2 <- read_excel(path, range = "D2:E4")
test <- read_excel(path, range = "G2:G9")

result <- input1 %>%
  mutate(row = row_number()) %>%
  left_join(input2, by = c("row" = "Item:")) %>%
  mutate(items = map2(`Items`, replace_na(`Repeats:`, 1), rep)) %>%
  unnest_longer(items) %>%
  select(items)

all.equal(result$items, test$Solution)
#> [1] TRUE
